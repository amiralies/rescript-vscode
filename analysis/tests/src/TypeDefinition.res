type variant = Foo | Bar

type record = {item: string}

let x = Foo
//  ^typ

let y = {item: "foo"}
//  ^typ

type obj = {"foo": string}

let obj: obj = {"foo": "bar"}
//  ^type

let f = r => r.item
//           ^typ

let g = v =>
  switch v {
  //     ^typ
  | Foo => "Foo"
  | Bar => "Bar"
  }
