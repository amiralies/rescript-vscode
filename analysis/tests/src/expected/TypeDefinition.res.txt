TypeDefinition tests/src/TypeDefinition.res 4:4
{"uri": "TypeDefinition.res", "range": {"start": {"line": 0, "character": 0}, "end": {"line": 0, "character": 24}}}

TypeDefinition tests/src/TypeDefinition.res 7:4
{"uri": "TypeDefinition.res", "range": {"start": {"line": 2, "character": 0}, "end": {"line": 2, "character": 28}}}

TypeDefinition tests/src/TypeDefinition.res 12:4
{"uri": "TypeDefinition.res", "range": {"start": {"line": 10, "character": 0}, "end": {"line": 10, "character": 26}}}

TypeDefinition tests/src/TypeDefinition.res 15:13
{"uri": "TypeDefinition.res", "range": {"start": {"line": 2, "character": 0}, "end": {"line": 2, "character": 28}}}

TypeDefinition tests/src/TypeDefinition.res 19:9
{"uri": "TypeDefinition.res", "range": {"start": {"line": 0, "character": 0}, "end": {"line": 0, "character": 24}}}

